src/components/Plans/Checkout/PurchaseSummary/InvoiceRecipients/index.js

```diff
@@ -17,42 +17,59 @@ export const InvoiceRecipients = InjectAppServices(
     const intl = useIntl();
     const _ = (id, values) => intl.formatMessage({ id: id }, values);
 
+    const updateInvoiceRecipients = useCallback(
+      (recipients) => {
+        const update = async () => {
+          const userEmail = appSessionRef.current.userData.user.email;
+          const recipientsToSend = recipients.length > 0 ? recipients : [userEmail];
+          const response = await dopplerBillingUserApiClient.updateInvoiceRecipients(
+            recipientsToSend,
+            selectedPlan,
+          );
+          if (response.success) {
+            setRecipients(recipientsToSend);
+          }
+          // TODO: define action to take when update fails
+        };
+        update();
+      },
+      [selectedPlan, dopplerBillingUserApiClient, appSessionRef],
+    );
+
+    const submitEditRecipients = useCallback(
+      (values) => {
+        updateInvoiceRecipients(values[fieldNames.editRecipients]);
+        setEdit(false);
+      },
+      [updateInvoiceRecipients],
+    );
+
     useEffect(() => {
       const fetchData = async () => {
         const userEmail = appSessionRef.current.userData.user.email;
-        const response = await dopplerBillingUserApiClient.getInvoiceRecipientsData();
-        if (response.success) {
-          setRecipients(response.value.recipients);
+        const { success, value } = await dopplerBillingUserApiClient.getInvoiceRecipientsData();
+        if (success) {
+          if (value.recipients?.length > 0) {
+            setRecipients(value.recipients);
+          } else {
+            submitEditRecipients({
+              [fieldNames.editRecipients]: [userEmail],
+            });
+          }
```

Comment:
I did not read the logic, but, in general, we are trying to reduce nesting. What do you think about this:

```suggestion
        const { success, value } = await dopplerBillingUserApiClient.getInvoiceRecipientsData();
        if (success && value.recipients?.length > 0) {
          setRecipients(value.recipients);
        } else if (success) {
          submitEditRecipients({
            [fieldNames.editRecipients]: [userEmail],
          });
```

Reply:
it's ok! better!